### DISCUSSION TITLE
Is the docker machine running on the runner manager

### DISCUSSION THREAD

**Discussion by lmilbaum** [npalm/terraform-aws-gitlab-runner#1020] [Q&A]
I'd like to understand the current architecture so that I can contribute to this project. 
Where is the docker machine running? The README diagrams are a bit confusing.

**Comment by kayman-mk** [comment]
Appreciated!

The docker+machine executor creates a new EC2 instance, installs Docker and executes the job within a container. So it is always a separate machine.

As far as I remember, the shell and docker executor run on the same machine as the Runner.

**Reply by lmilbaum** [reply]
Thanks for the clarification
